pkgs/run-sotest/src/run_sotest/__main__.py

```diff
@@ -54,13 +61,18 @@ def create_test_run(args):
     url = "{}/test_runs".format(args.sotest_url)
     files = {"config": open(args.sotest_config, "rb")}
     params = {
-        "boot_files_url": args.boot_files,
         "url": args.url,
         "user": args.user,
         "name": args.name,
         "priority": args.priority,
     }
 
+    # Check if boot files are a valid local path, assume URL otherwise
+    try:
+        files["boot_files"] = open(args.boot_files, "rb")
+    except:
+        params["boot_files_url"] = args.boot_files
```

Comment:
Is it a deliberate choice to use the same argument rather than two separate ones, of which exactly one must be given?

Reply:
Yes, it is; there are two reasons:

In SoTest's API we have two fields: `boot_files` for the direct upload, and `boot_files_url` for the URL only. Since in `run_sotest`, `boot_files` is used for the `boot_files_url`, naming for the new parameter might easily get confusing and difficult to remember. We could do something like `boot_files_path`, but that still would be a little weird. By using one parameter only, we circumvent this issue.

We'd have to make `boot_files` a non-required argument and then add a bit of "manual" argparsing to make sure that `boot_files` xor `boot_files_path` is specified. With the current solution, we don't need that because the argparser handles the validation for us.

I expected this choice to spark a discussion though and we can change it to two parameters if that seems to make more sense.

Reply:
Ok, I think from a "cleaning up old mistakes" perspective, we could introduce two new parameters and deprecate the old one, but then again it seems like a bit overkill for this simple tool. Just as with the positional parameter, I don't think it's a huge deal because the direct usage of the tool in the currently planned use cases might be rare.